### PR TITLE
BOAC-1841, rich-text editor for note details (ckeditor5)

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1841

Styles are customized according to [docs](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/ui/theme-customization.html). Injecting aria attributes in toolbar is not possible with `ckeditor5`. Work remains in assessing accessibility.

### CODE REVIEW

File: src/components/note/NewNoteModal.vue [hunk 1]
```diff
@@ -126,9 +135,12 @@
 </template>
 
 <script>
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
 import Util from '@/mixins/Util';
 import { createNote } from '@/api/notes';
 
+require('@/assets/styles/ckeditor-custom.css');
+
```
Comment (johncrossman) [inline src/components/note/NewNoteModal.vue]: Amended to move `ckeditor` CSS customizations to external file.

File: src/components/note/NewNoteModal.vue [hunk 2]
```diff
@@ -68,12 +68,21 @@
...
             <label for="create-note-body" class="input-label mt-3 mb-1">Note Details</label>
           </div>
-          <div>
-            <textarea id="create-note-body" v-model="body"></textarea>
+          <div id="note-details">
+            <textarea
+              v-if="false"
+              id="create-note-body"
+              v-model="body"
+              class="note-body-textarea"></textarea>
```
Comment (paulkerschen) [inline src/components/note/NewNoteModal.vue]: @johncrossman How does `v-if="false"` work here? Does it ever show?

Comment (johncrossman) [reply src/components/note/NewNoteModal.vue]: That's my "TODO: delete this"... which I'll todo now.

**Comment by johncrossman** [comment]
Amended. T is green.

**Comment by paulkerschen** [comment]
Those rich editors are why writers get starved, but I'll merge anyway.

**Pull request merged**